index.js

```diff
@@ -0,0 +1,408 @@
+   * Use to stop or remove any dynamic or background behavior such as events.
+   */
+  pause: function () {
+    this.el.removeEventListener("avatar-connect:show-modal", onShowModal);
+  },
+
+  /**
+   * Called when entity resumes.
+   * Use to continue or add any dynamic or background behavior such as events.
+   */
+  play: function () {
+    this.el.addEventListener(
+      "avatar-connect:show-modal",
+      this.onShowModal.bind(this)
+    );
+  },
+
+  /**
+   * Event handlers that automatically get attached or detached based on scene state.
+   */
+  events: {
+    // click: function (evt) { }
+  },
+
+  /**
+   * Custom helper functions
+   */
+  onShowModal: function (event) {
+    this.connector.enable();
+  },
+  onCloseModal: function (event) {
+    this.el.emit("avatar-connect:close-modal");
+  },
+  createAvatar: function (avatar) {
+    var avatarEl = document.createElement("a-entity");
+    // TODO: should be on and off turnable
+    avatarEl.setAttribute("avatar-connect-avatar", {
+      uri: avatar.uri,
+      format: avatar.format,
+      showPodest: true,
+    });
+    this.el.appendChild(avatarEl);
+  },
+});
+
+/**
+ * AvatarConnect Avatar component for A-Frame.
+ */
+AFRAME.registerComponent("avatar-connect-avatar", {
+  schema: {
+    uri: { default: "" },
+    format: { default: "glb" },
+    showPodest: { default: true },
+  },
+
+  /**
+   * Set if component needs multiple instancing.
+   */
+  multiple: false,
+
+  /**
+   * Called once when component is attached. Generally for initial setup.
+   */
+  init: async function () {
+    if (this.data.uri) {
+      await this.createAvatar();
+    } else {
+      this.el.emit("avatar-connect:error", "Avatar URI is empty");
+      console.error("URI is empty");
+    }
+  },
+
+  /**
+   * Called when component is attached and when component data changes.
+   * Generally modifies the entity based on the data.
+   */
+  update: function (oldData) {},
+
+  /**
+   * Called when a component is removed (e.g., via removeAttribute).
+   * Generally undoes all modifications to the entity.
+   */
+  remove: function () {},
+  /**
+   * Called on each scene tick.
+   */
+  // tick: function (t) { },
+
+  /**
+   * Called when entity pauses.
+   * Use to stop or remove any dynamic or background behavior such as events.
+   */
+  pause: function () {},
+
+  /**
+   * Called when entity resumes.
+   * Use to continue or add any dynamic or background behavior such as events.
+   */
+  play: function () {},
+
+  /**
+   * Event handlers that automatically get attached or detached based on scene state.
+   */
+  events: {
+    // click: function (evt) { }
+  },
+
+  /**
+   * Custom helper functions
+   */
+  createAvatar: async function () {
+    var avatarEl = document.createElement("a-entity");
+    // TODO: support more formats and providers
+    switch (this.data.format) {
+      case "glb":
+        const gender = await this.getRPMGender(this.data.uri);
```

Comment:
A quick note on this for RPM specifically - the bridge actually pre-fetches the RPM metadata and returns it in the `metadata` object in the bridge response, so you can use that instead!

### Example RPM Response:
```json
{
  "provider": "ready-player-me",
  "avatar": {
    "type": "humanoid",
    "format": "glb",
    "uri": "https://d1a370nemizbjq.cloudfront.net/855c8708-7410-4eeb-8d61-f217799c2b99.glb?v=83685"
  },
  "metadata": {
    "bodyType": "fullbody"
    "outfitGender": "masculine"
    "outfitVersion": 2
  }
}
```

Reply:
Ah, didn't know that. Great, that makes it easier. Will also prepare a new version for this.